url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/pl/cc/real/RealAgent.java

```diff
@@ -0,0 +1,323 @@
+		this(id,name,extension);
+		this.isPaused = isPaused;
+	}
+
+	public RealAgent(LoginCredentials login) {
+		this(login.username,login.username,login.exten);
+	}
+	
+	public void addAgentListener(RealAgentListener realAgentListener){
+		this.agentListenerList.add(realAgentListener);
+	}
+	
+	public void removeAgentListener(RealAgentListener realAgentListener){
+		this.agentListenerList.remove(realAgentListener);
+	}
+	
+	public RealAgent(OkWelcome selfInfo) {
+		this(selfInfo.getAgentNumber(), selfInfo.getName(), selfInfo.getExtension());
+	}
+
+	public Integer getId() {
+		return id;
+	}
+
+	public String getName() {
+		return name;
+	}
+
+	public String getExtension() {
+		return extension;
+	}
+
+	public int getPauseTime() {
+		return pauseTime;
+	}
+
+    public int getTimePoor() {
+        return timePoor;
+    }
+
+    public int getTimeBad() {
+        return timeBad;
+    }
+
+
+	public boolean setPaused(boolean paused, PauseType pauseType) {
+		return setPaused(paused, pauseType, PauseType.PAUSE_FOREVER, 0, 0);
+	}
+	
+	public boolean setPaused(boolean paused, PauseType pauseType, int pauseTime, int timePoor, int timeBad) {
+		if (this.isPaused!=paused || this.pauseType==null || !this.pauseType.equals(pauseType)){
+			 this.isPaused = paused;
+			 this.pauseType = pauseType;
+			 this.pauseTime = pauseTime;
+             this.timePoor = timePoor;
+             this.timeBad = timeBad;
+			agentListenerList.onPause(this, paused, pauseType, pauseTime, timePoor, timeBad);
+			lastStateChange=GregorianCalendar.getInstance().getTime();
+			return true;
+		}
+		return false;
+	}
+
+	
+	public Date getLastStateChange() {
+		return lastStateChange;
+	}
+
+	public int getTimeInLastState(){
+		if (connectedCall!=null) return (int) connectedCall.getSecElapsed();
+		return (int) ((GregorianCalendar.getInstance().getTime().getTime()-lastStateChange.getTime())/1000);
+	}
+	
+	public void setLastStateChange(Date lastStateChange) {
+		this.lastStateChange = lastStateChange;
+	}
+
+	public void onSpy() {
+		log.debug("UI onSpy: "+getName());
+		if (proxyTasks!=null){
+			proxyTasks.doSpyAgent(this);
+		} else {
+			log.debug("proxyTasks == null");
+		}
+	}
+	
+	public void doPause(boolean pause, PauseType pauseType) {
+		log.debug("UI doPause: "+getName());
+		if (proxyTasks!=null){
+			proxyTasks.doPauseAgent(this, pause, pauseType);
+		} else {
+			log.debug("proxyTasks == null");
+		}
+	}
+	
+	@Override
+	public void doPauseAdministrative(boolean pause) {
+		if (proxyTasks!=null){
+			proxyTasks.doPauseAgentAdministrative(this, pause);
+		} else {
+			log.debug("proxyTasks == null");
+		}
+	}
+	
+	public void onAgentNewCall(RealCall ringingCall, RealQueue queue) {
+		this.ringingCall = ringingCall;
+		this.connectedCall = null;
+		agentListenerList.onAgentNewCall(ringingCall, queue);
+	}
+
+	public void onCallLeave(RealCall callLeave, RealQueue queue) {
+		if (ringingCall==null) return;
+		if (callLeave.getUniqueID()==ringingCall.getUniqueID()){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.